### PR TITLE
feat: Add supported currencies to store model

### PR DESCRIPTION
**Pull request by sradevski**
I ultimately went with having a flat list of settings for the store. It wouldn't be very difficult to change it if we wish to do so, but for now this keeps the codebase simpler

### CODE REVIEW

File: packages/store/src/models/store.ts [hunk 1]
```diff
@@ -35,6 +35,9 @@ export default class Store {
   @Property({ columnType: "text", default: "Medusa Store" })
   name: string
 
+  @Property({ type: "array", default: "{}" })
```
Comment (adrien2p) [inline packages/store/src/models/store.ts:38]: Suggestion: i would default to [] as it is a valid jsonb value

**Review by adrien2p — APPROVED** [review]

File: packages/store/src/models/store.ts (hunk 1 above)
Comment (sradevski) [reply packages/store/src/models/store.ts:38]: @adrien2p why do we want it as jsonb, where Postgres has native support for primitive type arrays (and IIRC "{}" is how you represent an empty array, but maybe mikroorm does some magic if you pass []?)

Comment (adrien2p) [reply packages/store/src/models/store.ts:38]: My bad, mostly a habit, also jsonb areay are declare like this [] and with array you can either use {} or ARRAY[]. Feel free to discard

Comment (sradevski) [reply packages/store/src/models/store.ts:38]: @adrien2p No worries, I am always wondering if there is some magic happening that I should know of 😄

File: packages/medusa/src/api-v2/admin/stores/validators.ts [hunk 2]
```diff
@@ -41,6 +47,10 @@ export class AdminPostStoresStoreReq {
   @IsString()
   name?: string
 
+  @IsOptional()
+  @IsArray()
+  supported_currency_codes?: string[]
```
Comment (riqwan) [inline packages/medusa/src/api-v2/admin/stores/validators.ts:52]: Just a nit, but I think we can drop the `supported_` prefix, discardable.

**Review by riqwan — APPROVED** [review]
lgtm!

File: packages/medusa/src/api-v2/admin/stores/validators.ts (hunk 2 above)
Comment (sradevski) [reply packages/medusa/src/api-v2/admin/stores/validators.ts:52]: @riqwan I was a bit torn TBH, and I decided to add it to distinguish between "all currency codes" and the ones that the store decided to support.

**Pull request merged**